pkg/asset/agent/image/agentimage.go

```diff
@@ -86,9 +86,18 @@ func (a *AgentImage) Generate(ctx context.Context, dependencies asset.Parents) e
 		return fmt.Errorf("AgentWorkflowType value not supported: %s", agentWorkflow.Workflow)
 	}
 
+	var err error
+	if a.tmpPath, err = agentArtifacts.PrepareArtifacts(ctx); err != nil {
+		return err
+	}
+	defer func() {
```

Comment:
On a second check, I think this may not be required here.  Both the `AgentImage` and `AgentPxeFiles` remove the temp folder in their respecitve `PersistToFile` method, which is always invoked (even if the [Fetch fails](https://github.com/openshift/installer/blob/cf138a1f61c8ff96c970011f5bf502d23906c13f/pkg/asset/store/assetsfetcher.go#L52))